src/main/java/org/jenkinsci/account/BoardElection.java

```diff
@@ -30,64 +25,272 @@
+    private final String resultsLogDir;
+    private final String seats;
+    private ArrayList<HashMap<String,String>> result;
+    private ArrayList<String> rawResult;
+    private final int seniority;
+    private int totalVotes;
 
     public BoardElection(Application application, Parameters params) throws Exception {
-        this.app = application;
+        app = application;
+        seniority = params.seniority();
+        result = new ArrayList<HashMap<String, String>>();
         final SimpleDateFormat format = new SimpleDateFormat("yyyy/MM/dd");
         open = format.parse(params.electionOpen());
         close = format.parse(params.electionClose());
         candidates = params.electionCandidates().split(",");
-        log = params.electionLogDir() + "/" + params.electionClose().replaceAll("/","") + ".yaml";
-        File f = new File(log);
-        f.createNewFile();
+        electionLogDir = params.electionLogDir();
+        seats = params.seats();
+        rawResult = new ArrayList<String>();
+
+        LocalDate date = close.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+        resultsLogDir = electionLogDir + "/" + date.format(DateTimeFormatter.BASIC_ISO_DATE);
+
+        File dir = new File(resultsLogDir);
+        dir.mkdirs();
     }
 
-    @RequirePOST
-    public HttpResponse doVote(@QueryParameter String vote) throws NamingException, IOException {
+    private String messageDigest(String s) throws NoSuchAlgorithmException {
+        MessageDigest md = MessageDigest.getInstance("MD5");
+        md.update(s.getBytes());
+        byte[] digest = md.digest();
+        return DatatypeConverter.printHexBinary(digest);
+    }
+
+    public HttpResponse doResults() throws NamingException, IOException {
+        return HttpResponses.forwardToView(this,"results.jelly");
+    }
 
-        final Myself user = Myself.current();
-        if (user == null) {
-            throw new UserError("Need to be authenticated");
+    public HttpResponse doResult(@QueryParameter String election) throws NamingException, IOException {
+
+        if ( election.isEmpty() || election.equals(".") || election.equals("..")){
+            return new HttpRedirect("/election/results");
+        }
+
+        File directory = new File( electionLogDir + "/" + election );
+        if (!directory.isDirectory() || !directory.exists()){
+            return new HttpRedirect("/election/results");
         }
 
-        if (!isOpen())  {
-            throw new UserError("Election is closed");
+        if (!isResultExist(election) || isOpen(election)){
+            return new HttpRedirect("/election/ongoing");
+        }
+        else{
+            parseRawResult( election );
+            return HttpResponses.forwardToView(this,"result.jelly");
+        }
+    }
+
+    public HttpResponse doUpdateResult(@QueryParameter String election) throws IOException {
+        if (!isOpen(election)){
+            countVotes(election);
```

Comment:
I'm not entirely clear on how this is going to be called, but it has me concerned because this action is going to re-process a CSV file, rewrite a new file, and then run a python process.

Reply:
This function can only be called by an admin user, the condition is inside results.jelly
-> https://github.com/olblak/account-app/blob/fcb5756d85db4ae750d88ae61a67e8a9059c6f56/src/main/resources/org/jenkinsci/account/BoardElection/results.jelly

Reply:
@olblak It's still bound to a URL, and doesn't perform permission checks.

Also, no `@RequirePOST` for CSRF goodness on top.

Reply:
In "short":
For a regular use, the workflow is as follows.
doVote():
  Write a user vote into a file.
  Ex: md5("userid").csv
    |   
    |   
    v 
doResults():
  List all elections and provide a link for each of them to display one election result.
    |   
    |   
    v   
doResult():
  Display one election result based on parseRawResult().

        parseRawResult():
          Read "results.stv" and display contents.

---
Admin workflow

An admin follows the same workflow than a classical user excepted that during the doResults() steps,
it can call the function doUpdateResults which read all user vote files to generate one big file called "votes.csv" then  
run the python script on it and write the result to "results.stv".

Reply:
@daniel-beck Ho I see your point

Reply:
@daniel-beck Ho indeed